hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/RecoverableZooKeeper.java

```diff
@@ -110,6 +112,23 @@ public static RecoverableZooKeeper connect(Configuration conf, String ensemble,
    */
   public static RecoverableZooKeeper connect(Configuration conf, String ensemble, Watcher watcher,
     final String identifier) throws IOException {
+    return connect(conf, ensemble, watcher, identifier, null);
```

Comment:
Why passing null instead of create an empty ZKClientConfig? In the above protected method, we create an empty ZKClientConfig instance...

Reply:
Technically it would not make a difference, because the ZooKeeper code we end up calling would just create a new ZKClientConfig when a null is passed.
I implemented it like this to signal where the config should be modified if needed. I pass a null for use cases where the config was not passed and we expect to use the default. But I pass the instance where I expect it to be modified.

Reply:
`ZKClientConfig` param is marked as Nullable in ZK code, so I think it's fine to pass null here. However you can also use the method overload _without_ client config parameter if you don't want to customize it.